src/garage/torch/algos/td3.py

```diff
@@ -0,0 +1,429 @@
+                self._actor_loss.detach())
+
+    def _evaluate_policy(self):
+        """Evaluate the performance of the policy via deterministic rollouts.
+
+        Statistics such as (average) discounted return and success rate are
+            recorded.
+
+        Returns:
+            TrajectoryBatch: Evaluation trajectories, representing the best
+                current performance of the algorithm.
+
+        """
+        return obtain_evaluation_episodes(
+            self.exploration_policy,
+            self._eval_env,
+            num_eps=self._num_evaluation_episodes,
+            deterministic=False)
+
+    def _update_network_parameters(self):
+        """Update parameters in actor network and critic networks."""
+        for target_param, param in zip(self._target_qf_1.parameters(),
+                                       self._qf_1.parameters()):
+            target_param.data.copy_(target_param.data * (1.0 - self._tau) +
+                                    param.data * self._tau)
+
+        for target_param, param in zip(self._target_qf_2.parameters(),
+                                       self._qf_2.parameters()):
+            target_param.data.copy_(target_param.data * (1.0 - self._tau) +
+                                    param.data * self._tau)
+
+        for target_param, param in zip(self._target_policy.parameters(),
+                                       self.policy.parameters()):
+            target_param.data.copy_(target_param.data * (1.0 - self._tau) +
+                                    param.data * self._tau)
+
+    def _log_statistics(self):
+        """Output training statistics to dowel such as losses and returns."""
+        tabular.record('Policy/AveragePolicyLoss',
+                       np.mean(self._episode_policy_losses))
+        tabular.record('QFunction/AverageQFunctionLoss',
+                       np.mean(self._episode_qf_losses))
+        tabular.record('QFunction/AverageQ', np.mean(self._epoch_qs))
+        tabular.record('QFunction/MaxQ', np.max(self._epoch_qs))
+        tabular.record('QFunction/AverageAbsQ',
+                       np.mean(np.abs(self._epoch_qs)))
+        tabular.record('QFunction/AverageY', np.mean(self._epoch_ys))
+        tabular.record('QFunction/MaxY', np.max(self._epoch_ys))
+        tabular.record('QFunction/AverageAbsY',
+                       np.mean(np.abs(self._epoch_ys)))
+
+    @property
+    def networks(self):
+        """Return all the networks within the model.
+
+        Returns:
+            list: A list of networks.
+
+        """
+        return [
+            self.policy, self._qf_1, self._qf_2, self._target_policy,
+            self._target_qf_1, self._target_qf_2
+        ]
+
+    def to(self, device=None):
+        """Put all the networks within the model on device.
+
+        Args:
+            device (str): ID of GPU or CPU.
+
+        """
+        if torch.cuda.is_available():
```

Comment:
I don't think the other algorithms' `.to()` methods do this. Can you move this logic to create a function such as `garage.torch.prefer_gpu()`